subsys/bootloader/bl_crypto/bl_crypto.c

```diff
@@ -116,36 +116,27 @@ int bl_sha256_verify(const u8_t *data, u32_t data_len, const u8_t *expected)
 	return verify_truncated_hash(data, data_len, expected, CONFIG_SB_HASH_LEN, true);
 }
 
-__ext_abi(struct bl_rot_verify_abi, bl_rot_verify_abi) = {
-	.header = FW_INFO_ABI_INIT(BL_ROT_VERIFY_ABI_ID,
-				CONFIG_BL_ROT_VERIFY_ABI_FLAGS,
-				CONFIG_BL_ROT_VERIFY_ABI_VER,
-				sizeof(struct bl_rot_verify_abi)),
-	.abi = {
+
+__ext_abi(BL_ROT_VERIFY_ABI_ID, CONFIG_BL_ROT_VERIFY_ABI_FLAGS,
```

Comment:
I like the way this is going, but there is still a lot of repetition in these definitions. Are these left here to have some more flexibility, or could we assume that the kconfig options has the correct names (if defined by kconfig templates) and make it :

```
__ext_abi(BL_ROT_VERIFY, bl_rot_verify_abi)
```
?

Or would that reduce usability as the error codes would be very hard to decode by the user when creating new ext-abis?

Thinking out loud here :)

Reply:
Hadn't thought of that. I like how it turned out :).